lib/ohai/plugins/ec2.rb

```diff
@@ -69,11 +50,40 @@ def has_ec2_dmi?
     end
   end
 
+  # looks for a xen UUID that starts with ec2
+  # this gets us detection of Linux HVM and Paravirt hosts
+  def has_ec2_xen_uuid?
+    if ::File.exist?("/sys/hypervisor/uuid")
+      if ::File.read("/sys/hypervisor/uuid") =~ /^ec2/
+        Ohai::Log.debug("ec2 plugin: has_ec2_xen_uuid? == true")
+        return true
+      end
+    else
+      Ohai::Log.debug("ec2 plugin: has_ec2_xen_uuid? == false")
+      return false
+    end
+  end
+
+  # looks for the Amazon.com Organization in Windows Kernel data
+  # this gets us detection of Windows systems
+  def has_amazon_org?
+    begin
+      # detect an Organization of 'Amazon.com'
+      if kernel[:os_info][:organization] =~ /Amazon/
+        Ohai::Log.debug("ec2 plugin: has_amazon_org? == true")
+        true
+      end
+    rescue NoMethodError
```

Comment:
I'm slightly uncomfortable catching the nonexistance of `kernel[:os_info][:organization]` with an exception. If it's set to not Amazon, then you'll also miss the logging.